modules/signals/src/signal-store.ts

```diff
@@ -1355,12 +1355,18 @@ export function signalStore(
         getInitialInnerStore()
       );
       const { stateSignals, props, methods, hooks } = innerStore;
-      const storeMembers = { ...stateSignals, ...props, ...methods };
+      const storeMembers: Record<string | symbol, unknown> = {
+        ...stateSignals,
+        ...props,
+        ...methods,
+      };
 
       (this as any)[STATE_SOURCE] = innerStore[STATE_SOURCE];
 
-      for (const key in storeMembers) {
-        (this as any)[key] = storeMembers[key];
+      for (const key of Reflect.ownKeys(storeMembers)) {
+        if (typeof key === 'string' || typeof key === 'symbol') {
+          (this as any)[key] = storeMembers[key];
+        }
```

Comment:
what's the reason for adding this `if`? is there any other type of key other than string or symbol that should be ignored?
```suggestion
        (this as any)[key] = storeMembers[key];
```

Reply:
`key` in this case is of type `string | symbol` and TypeScript doesn't allow an access with that type. That's why I had to apply type narrowing.

Litte bit strange, since I've never experienced that before.

Reply:
I just checked and it seems to work on my machine 😅 

![1](https://github.com/user-attachments/assets/75f109a4-8cf2-401c-831c-46062aebea8a)

Anyway, instead of adding additional LOC that will end up in the final bundle unnecessarily in order to satisfy TS, I suggest using type casting if it doesn't work without `if (typeof key === 'string' ...`